src/main/java/ch/baloise/corellia/api/doc/JsonFormatObjectEnumConverter.java

```diff
@@ -0,0 +1,89 @@
+package ch.baloise.corellia.api.doc;
+
+import com.fasterxml.jackson.annotation.JsonFormat;
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.fasterxml.jackson.databind.JavaType;
+import io.swagger.v3.core.converter.AnnotatedType;
+import io.swagger.v3.core.converter.ModelConverter;
+import io.swagger.v3.core.converter.ModelConverterContext;
+import io.swagger.v3.core.util.Json;
+import io.swagger.v3.oas.models.media.ObjectSchema;
+import io.swagger.v3.oas.models.media.Schema;
+import io.swagger.v3.oas.models.media.StringSchema;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.AccessibleObject;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * A {@link ModelConverter} dedicated for enum using {@link JsonValue} annoation on a getter.
+ * This is then used to create a specification for a enum in the form of ENUM_INSTANCE_NAME: VALUE.
+ */
+public class JsonFormatObjectEnumConverter implements ModelConverter {
+
+  private JsonPropertyDescriptionResolver jsonPropertyDescriptionResolver = new JsonPropertyDescriptionResolver(Json.mapper());
+
+  @Override
+  public Schema resolve(AnnotatedType type, ModelConverterContext context, Iterator<ModelConverter> chain) {
```

Comment:
This method does a lot, could we refactor or extract stuff into separate methods?

Reply:
Done, see 498ac2b